exonum/src/blockchain/schema.rs

```diff
@@ -384,40 +384,41 @@ enum IndexTag {
 /// Normalized coordinates of the index in the `state_hash_aggregator` table.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
 pub struct IndexCoordinates {
-    tag: u16,
-    group_id: u32,
+    origin: u16,
+    schema_id: u32,
     index_id: u16,
 }
 
 impl IndexCoordinates {
-    /// Creates index coordinated for the index with the specified owner and identifier.
-    pub fn new(owner: IndexOwner, index_id: u16) -> Self {
+    /// Creates index coordinates for the index with the specified schema origin
+    /// and index identifier.
+    pub fn new(schema_origin: SchemaOrigin, index_id: u16) -> Self {
         Self {
-            tag: owner.tag() as u16,
-            group_id: owner.group_id(),
+            origin: schema_origin.origin() as u16,
+            schema_id: schema_origin.schema_id(),
             index_id,
         }
     }
 
     // TODO Write more meaningful description [ECR-3824]
-    /// For the given index owner, returns a list of the index coordinates that match the corresponding
-    /// object hashes of the indices.
+    /// For the given index owner, returns a list of the index coordinates that match the
+    /// corresponding object hashes of the indices.
     pub fn locate(
-        owner: IndexOwner,
+        schema_origin: SchemaOrigin,
         object_hashes: impl IntoIterator<Item = Hash>,
     ) -> impl IntoIterator<Item = (IndexCoordinates, Hash)> {
         object_hashes
             .into_iter()
             .enumerate()
-            .map(move |(id, hash)| (owner.coordinate_for(id as u16), hash))
+            .map(move |(id, hash)| (schema_origin.coordinate_for(id as u16), hash))
     }
 
-    /// Returns a membership for this index.
-    pub fn owner(self) -> IndexOwner {
-        match self.tag {
-            0 => IndexOwner::Core,
-            2 => IndexOwner::Runtime(self.group_id),
-            3 => IndexOwner::Service(self.group_id),
+    /// Returns a schema origin for this index.
+    pub fn schema_origin(self) -> SchemaOrigin {
+        match self.origin {
+            0 => SchemaOrigin::Core,
+            2 => SchemaOrigin::Runtime(self.schema_id),
```

Comment:
Off-topic: interesting, where did we lose `1`?

Reply:
'1' is an origin of the `Dispatcher` schema.

Reply:
I guess it should be mentioned somewhere.

Reply:
By the way, it will be added here after implementing this issue ECR-3810